### ISSUE TITLE
fdt_rs::base prop().node() iteration appears to be bugged.

### ISSUE THREAD

**Issue by spwilson2** [rs-embedded/fdt-rs#1]
The following test fails in the "Device tree memory node missing 'reg' prop." expect clause using the riscv-5 device tree.

```rust
#[test]
fn get_memory_prop_node() {
    unsafe {
        let blob = DevTree::new(FDT).unwrap();

        let mem_prop = blob.props().find(|p| Ok(p.name()? == "device_type" && p.str()? == "memory"))
            .unwrap()
            .expect("Unable to find memory node.");
        let mem_node = mem_prop.node();
        let mem_reg_prop = mem_node.props()
            .find(|p| Ok(p.name()? == "reg"))
            .unwrap()
            .expect("Device tree memory node missing 'reg' prop.");
}
```

**Comment by spwilson2** [comment]
Note that the `fdt_rs::index` version of this works fine.

**Comment by spwilson2** [comment]
Fixed by ea35a72270fece3fd19ec756fc670b9840e9a8bc
